### PR TITLE
Stop waiting for pipeline update completion if there are no updates

### LINKED ISSUE #465: Pipeline update can't complete if trying to update an unchanged pipeline
Stuck at `Proposing infrastructure changes for the pipeline` when trying to deploy an unchanged pipeline.

However, it won't get stuck if the pipeline already got updated with a change and then do an update with no change.

### PR DESCRIPTION
**Pull request by efekarakus**
Fixes #465 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/pipeline_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	testCases := map[string]struct {
+		in *deploy.CreatePipelineInput
+
+		mockDescribeStacks                              func(t *testing.T, in *cloudformation.DescribeStacksInput) (*cloudformation.DescribeStacksOutput, error)
+		mockCreateChangeSet                             func(t *testing.T, in *cloudformation.CreateChangeSetInput) (*cloudformation.CreateChangeSetOutput, error)
+		mockWaitUntilChangeSetCreateCompleteWithContext func(t *testing.T, in *cloudformation.DescribeChangeSetInput) error
+		mockDescribeChangeSet                           func(t *testing.T, in *cloudformation.DescribeChangeSetInput) (*cloudformation.DescribeChangeSetOutput, error)
+		mockDeleteChangeSet                             func(t *testing.T, in *cloudformation.DeleteChangeSetInput) (*cloudformation.DeleteChangeSetOutput, error)
+
+		wantedError error
```
Comment (hencrice) [inline internal/pkg/deploy/cloudformation/pipeline_test.go]: nit: seems like `wantedError` is not used

**Comment by sonofachamp** [comment]
Looks good, the only question I have is do we want to inform users of scenarios such as "no changes", or do we want it to look like everything succeeded?

I think either way is fine, but I could see the explicitness being helpful to the user so that they know nothing changed. That wouldn't necessarily have to manifest itself as an "error", but that could also be an option.

File: internal/pkg/deploy/cloudformation/pipeline_test.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/pipeline_test.go]: it's used below within `Run`, I think as we add testCases to this unit test we'll have to use it :D

**Comment by efekarakus** [comment]
> Looks good, the only question I have is do we want to inform users of scenarios such as "no changes", or do we want it to look like everything succeeded?
> 
> I think either way is fine, but I could see the explicitness being helpful to the user so that they know nothing changed. That wouldn't necessarily have to manifest itself as an "error", but that could also be an option.

Yeah that's a good question, maybe providing more info would be nice. I'll merge this but we can do it as a UX improvement, I think our other commands could benefit of that information too :)

**Pull request merged**